src/autocomplete/databases/trino/grammar/TrinoParser.g4

```diff
@@ -286,8 +298,8 @@ joinCriteria
     ;
 
 sampledRelation
-    : patternRecognition (TABLESAMPLE_ sampleType LPAREN_ percentage = expression RPAREN_)?
-    | tableIdentifier
+    : tableReference
+    | patternRecognition (TABLESAMPLE_ sampleType LPAREN_ percentage = expression RPAREN_)?
```

Comment:
They need to be reversed, because patterRecognition can include almost anything, and we want to have `tableReference` in a priority